### PR TITLE
86Box: init at 3.11

### PR DESCRIPTION
**Pull request by jchv**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/86box/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    glib
+    openal
+    pcre2
+    jack2
+    libpcap
+    qt5.qtbase
+    qt5.qttools
+  ] ++ lib.optional (stdenv.isLinux) alsa-lib
+    ++ lib.optional (enableRtMidi) rtmidi
+    ++ lib.optional (enableVncRenderer) libvncserver;
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/86box/default.nix:45]:
```suggestion
  ] ++ lib.optional stdenv.isLinux alsa-lib
    ++ lib.optional enableRtMidi rtmidi
    ++ lib.optional enableVncRenderer libvncserver;
```
those are optional

File: pkgs/applications/emulators/86box/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  cmakeFlags = []
+    ++ lib.optional (!enableDynarec) "-DDYNAREC=OFF"
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/86box/default.nix:48]:
```suggestion
  cmakeFlags = lib.optional (!enableDynarec) "-DDYNAREC=OFF"
```

File: pkgs/applications/emulators/86box/default.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+, discord-gamesdk, libpcap
+
+, enableDynarec ? with stdenv; isx86_32 || isx86_64 || isAarch64 || isAarch32
+, enableNewDynarec ? with stdenv; enableDynarec && (isAarch64 || isAarch32)
+, enableFluidsynth ? true
+, enableRtMidi ? true
+, enableVncRenderer ? false
+, enableMinitrace ? false
+, enableGdbStub ? false
+, unfreeEnableDiscord ? false
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/86box/default.nix:12]: Do we really need all of this option?

File: pkgs/applications/emulators/86box/default.nix (hunk 1 above)
Comment (jchv) [reply pkgs/applications/emulators/86box/default.nix]: OK, I've removed superfluous parens in my lib.optional calls.

File: pkgs/applications/emulators/86box/default.nix (hunk 2 above)
Comment (jchv) [reply pkgs/applications/emulators/86box/default.nix]: Done.

File: pkgs/applications/emulators/86box/default.nix (hunk 3 above)
Comment (jchv) [reply pkgs/applications/emulators/86box/default.nix]: Sorry, I was genuinely unsure where to draw the line, so I just included all of the configurable options. I've removed most of the options and just kept the ones that are less trivial.

File: pkgs/applications/emulators/86box/default.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  # Do not wrap twice.
+  dontWrapQtApps = true;
+
+  meta = with lib; {
+    description = "Emulator of x86-based machines based on PCem.";
+    homepage = "https://86box.net/";
+    license = with licenses; [ gpl2Only ] ++ optional unfreeEnableDiscord unfree;
+    maintainers = [ maintainers.jchw ];
+    platforms = platforms.linux ++ platforms.darwin;
```
[the remark this answers is not in the document]
Comment (jchv) [reply pkgs/applications/emulators/86box/default.nix]: Thanks, applied. I did test this on macOS and it ran, so I'm not sure what I did wrong, but oh well.

File: pkgs/applications/emulators/86box/default.nix [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config, makeWrapper, freetype, SDL2
+, glib, pcre2, openal, rtmidi, fluidsynth, jack2, alsa-lib, qt5, libvncserver
+, discord-gamesdk, libpcap
+
+, enableDynarec ? with stdenv; isx86_32 || isx86_64 || isAarch64 || isAarch32
+, enableNewDynarec ? with stdenv; enableDynarec && (isAarch64 || isAarch32)
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/86box/default.nix:6]:
```suggestion
, enableNewDynarec ? enableDynarec && stdenv.isAarch
```

File: pkgs/applications/emulators/86box/default.nix [hunk 6]
```diff
@@ -0,0 +1,74 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config, makeWrapper, freetype, SDL2
+, glib, pcre2, openal, rtmidi, fluidsynth, jack2, alsa-lib, qt5, libvncserver
+, discord-gamesdk, libpcap
+
+, enableDynarec ? with stdenv; isx86_32 || isx86_64 || isAarch64 || isAarch32
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/86box/default.nix:5]:
```suggestion
, enableDynarec ? with stdenv; isx86 || isx86 || isAarch
```

File: pkgs/applications/emulators/86box/default.nix (hunk 5 above)
Comment (jchv) [reply pkgs/applications/emulators/86box/default.nix]: Neither `stdenv.isAarch` or `stdenv.isx86` seem to exist; apparently, those attrs are only available on `stdenv.hostPlatform`, so I moved to using `hostPlatform` instead.

File: pkgs/applications/emulators/86box/default.nix (hunk 6 above)
Comment (jchv) [reply pkgs/applications/emulators/86box/default.nix]: Same as above, I did:
```nix
, enableDynarec ? with stdenv.hostPlatform; isx86 || isAarch
```
instead.

File: pkgs/applications/emulators/86box/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/applications/emulators/86box/default.nix]: Oh, yes, good catch but I think you got my general direction.

**Comment by SuperSandro2000** [comment]
@ofborg build _86Box

**Pull request merged**